ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -647,4 +674,46 @@ - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
     return _shouldResignFirstResponder;
 }
 
+- (void)forgotPasscodeTapped
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(forgotPasscodeTapped:)])
+    {
+        [self.passcodeDelegate forgotPasscodeTapped:_forgotPasscodeButton];
+    }
+}
+
+- (BOOL) hasForgotPasscode
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(hasForgotPasscode)])
+    {
+        return [self.passcodeDelegate hasForgotPasscode];
+    }
+    return NO;
+}
+
+- (NSString*) forgotPasscodeButtonText
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(textForForgotPasscode)])
+    {
+        return [self.passcodeDelegate textForForgotPasscode];
+    }
+    return @"";
```

Comment:
That is why I was avoiding it in the first place.  In my opinion, it seems wrong to add a user facing string to ResearchKit, that isn't localized.  I can do as you suggested.